### PR TITLE
refactor/memline

### PR DESCRIPTION
**Pull request by lewis6991**
- refactor(memline): resolve clint errors
- refactor(memline): reduce local var scope

### CODE REVIEW

**Comment by clason** [comment]
rebase?

File: src/nvim/memline.c [hunk 1]
```diff
@@ -2805,14 +2721,15 @@ static bhdr_T *ml_find_line(buf_T *buf, linenr_T lnum, int action)
     return NULL;
   }
 
-  bnum = 1;                         // start at the root of the tree
-  page_count = 1;
-  low = 1;
-  high = buf->b_ml.ml_line_count;
+  blocknr_T bnum2;
+  blocknr_T bnum = 1;                         // start at the root of the tree
```
Comment (zeertzjq) [inline src/nvim/memline.c:2725]: Putting `bnum2` before `bnum` looks a bit strange.

File: src/nvim/memline.c [hunk 2]
```diff
@@ -2413,23 +2357,25 @@ static int ml_delete_int(buf_T *buf, linenr_T lnum, bool message)
...
   }
 
+  bhdr_T *hp;
   if ((hp = ml_find_line(buf, lnum, ML_DELETE)) == NULL) {
     return FAIL;
   }
 
-  dp = hp->bh_data;
+  DATA_BL *dp = hp->bh_data;
   // compute line count before the delete
```
Comment (zeertzjq) [inline src/nvim/memline.c:2371]: The deleted comment can be moved here and it does make code clearer
```suggestion
  // compute line count (number of entries in block) before the delete
```

File: src/nvim/memline.c [hunk 3]
```diff
@@ -3100,24 +3013,21 @@ char_u *makeswapname(char_u *fname, char_u *ffname, buf_T *buf, char_u *dir_name
...
+  char *tail = path_tail((char *)fname);
 
   if (dname[0] == '.' && dname[1] == NUL) {
     retval = xstrdup(fname);
   } else if (dname[0] == '.' && vim_ispathsep(dname[1])) {
     if (tail == fname) {            // no path before file name
       retval = concat_fnames(dname + 2, tail, true);
     } else {
-      save_char = (uint8_t)(*tail);
+      int save_char = *tail;
```
Comment (zeertzjq) [inline src/nvim/memline.c:3026]: Better change this to `char` change remove the cast below.
```suggestion
      char save_char = *tail;
```

File: src/nvim/memline.c [hunk 4]
```diff
@@ -2857,15 +2774,16 @@ static bhdr_T *ml_find_line(buf_T *buf, linenr_T lnum, int action)
...
     top = ml_add_stack(buf);  // add new entry to stack
-    ip = &(buf->b_ml.ml_stack[top]);
+    infoptr_T *ip = &(buf->b_ml.ml_stack[top]);
     ip->ip_bnum = bnum;
     ip->ip_low = low;
     ip->ip_high = high;
     ip->ip_index = -1;                  // index not known yet
 
-    dirty = false;
+    int dirty = false;
```
Comment (zeertzjq) [inline src/nvim/memline.c:2783]:
```suggestion
    bool dirty = false;
```

**Review by zeertzjq — APPROVED** [review]

File: src/nvim/memline.c [hunk 5]
```diff
@@ -3100,23 +3013,20 @@ char_u *makeswapname(char_u *fname, char_u *ffname, buf_T *buf, char_u *dir_name
 /// @param dname  don't use "dirname", it is a global for Alpha
 char *get_file_in_dir(char *fname, char *dname)
 {
-  char *t;
-  char *tail;
   char *retval;
-  int save_char;
 
-  tail = path_tail(fname);
+  char *tail = path_tail((char *)fname);
```
Comment (zeertzjq) [inline src/nvim/memline.c:3018]: Forgot about this one
```suggestion
  char *tail = path_tail(fname);
```

File: src/nvim/memline.c [hunk 6]
```diff
@@ -3002,7 +2916,7 @@ int resolve_symlink(const char *fname, char *buf)
       return FAIL;
     }
 
-    ret = (int)readlink(tmp, buf, MAXPATHL - 1);
+    int ret = (int)readlink((char *)tmp, (char *)buf, MAXPATHL - 1);
```
Comment (zeertzjq) [inline src/nvim/memline.c:2919]:
```suggestion
    int ret = (int)readlink(tmp, buf, MAXPATHL - 1);
```

**Pull request merged**